Transfer Cache updated with new Fetch Response API options and fixed HttpErrorResponse omissions

SkyZeroZx opened the pull request:
This PR adds the following commits


fix(http): handle missing redirected and responseType in HttpErrorResponse 

feat(http): add redirected and responseType params to HttpTestingController.expectOne flush 

feat(http): add redirected and responseType to transferCache with withHttpTransferCacheOptions

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

SkyZeroZx committed:
fix(http): handle missing redirected and responseType in HttpErrorResponse

Fixes omission of redirected and responseType from the Fetch API in HttpErrorResponse when the promise fails or when an error occurs while parsing the response body.

SkyZeroZx committed:
feat(http): add redirected and responseType params to HttpTestingController.expectOne flush

Add support for `redirected` and `responseType` parameters in the `flush` method of `HttpTestingController.expectOne` to properly validate Fetch Response API behavior in unit tests.

JeanMeche reviewed (approved):
reviewed-for: public-api

SkyZeroZx commented:
> Do you mind rebasing to fix the conflict ?

I'm already resolving the conflicts

JeanMeche commented on packages/common/http/src/transfer_cache.ts:
```diff
@@ -235,6 +245,8 @@ export function transferCacheInterceptorFn(
             [REQ_URL]: requestUrl,
             [RESPONSE_TYPE]: req.responseType,
+            [REDIRECTED]: event.redirected,
+            [EVENT_RESPONSE_TYPE]: event.responseType,
```
I would set those props only is the existed in the first place. (so we don't over load transfer state)

SkyZeroZx replied:
Good catch

SkyZeroZx replied:
Done

SkyZeroZx committed:
feat(http): add redirected and responseType to transferCache with withHttpTransferCacheOptions

Add `redirected` and `responseType` support to the transfer cache when using `withHttpTransferCacheOptions`, ensuring these fields are included in the cached transfer state.
